spice_ev/costs.py

```diff
@@ -445,12 +442,21 @@ def calculate_costs(strategy, voltage_level, interval,
         # power of flexible load:
         power_flex_load_list = get_flexible_load(power_grid_supply_list, power_fix_load_list)
 
+        # positive deviation from schedule concerning grid supply (not feed-in):
+        power_grid_supply_schedule_list = [max(v, 0) for v in power_schedule_list]
+        pos_deviation_grid_supply_list = [max(schedule_power - power_grid_supply_list[i], 0)
+                                          for i, schedule_power
+                                          in enumerate(power_grid_supply_schedule_list)]
+
         # prices:
         # set a suitable utilization time in order to use prices for grid friendly charging
         utilization_time_per_year_flex = UTILIZATION_TIME_PER_YEAR_EC
         commodity_charge_flex, capacity_charge_flex, fee_type = \
             find_prices(price_sheet_json, strategy, voltage_level, utilization_time_per_year_flex,
                         energy_supply_per_year, UTILIZATION_TIME_PER_YEAR_EC)
+        # charge for deviation from schedule:
+        schedule_deviation_charge = price_sheet["strategy_related_cost_parameters"][
+            "schedule"]["deviation_charge"]
```

Comment:
seems weird to split these lines (see also line 425). Suggestion: define an intermediate variable and then use that instead:
```
schedule_charges = price_sheet["strategy_related_cost_parameters"]["schedule"]
reduction_commodity_charge = schedule_charges["reduction_of_commodity_charge"]
...
schedule_deviation_charge = schedule_charges["deviation_charge"]
```